File: app/src/main/java/com/luobo/wanandroid/ui/home/aq/AqDiffUtil.java
package com.luobo.wanandroid.ui.home.aq;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

public class AqDiffUtil extends DiffUtil.ItemCallback<AqResponse.DatasBean> {
    @Override
    public boolean areItemsTheSame(@NonNull AqResponse.DatasBean oldItem, @NonNull AqResponse.DatasBean newItem) {
        return false;
    }

    @Override
    public boolean areContentsTheSame(@NonNull AqResponse.DatasBean oldItem, @NonNull AqResponse.DatasBean newItem) {
        return oldItem.getLink().equals(newItem.getLink());
    }
}

File: app/src/main/java/com/luobo/wanandroid/ui/home/aq/AqDiffUtil.kt
package com.luobo.wanandroid.ui.home.aq

import androidx.recyclerview.widget.DiffUtil
import com.luobo.wanandroid.ui.home.aq.AqResponse

internal class AqDiffUtil : DiffUtil.ItemCallback<AqResponse.DatasBean>() {
    override fun areItemsTheSame(oldItem: AqResponse.DatasBean, newItem: AqResponse.DatasBean): Boolean {
        return oldItem === newItem
    }

    override fun areContentsTheSame(oldItem: AqResponse.DatasBean, newItem: AqResponse.DatasBean): Boolean {
        return oldItem.id == newItem.id
    }
}